src/dex/curve-v1-factory/curve-v1-factory.ts

```diff
@@ -1115,11 +1141,21 @@ export class CurveV1Factory
       hexZeroPad(hexlify(blockNumber), 16),
     ]);
 
+    let wrapFlag = 0;
+
+    if (this.needWrapNative && isETHAddress(srcToken)) {
+      wrapFlag = 1; // wrap src eth
+    } else if (!this.needWrapNative && isETHAddress(srcToken)) {
+      wrapFlag = 3; // send eth
```

Comment:
better to specify that this flag just add msg.value to router call

Reply:
updated